gobblin-runtime/src/main/java/org/apache/gobblin/runtime/Task.java

```diff
@@ -130,14 +129,14 @@
   private static final String TASK_STATE = "taskState";
   private static final String FAILED_TASK_EVENT = "failedTask";
 
-  private final String jobId;
-  private final String taskId;
+  protected String jobId;
```

Comment:
Leave these as final and add a new constructor that can be used to initialize this to avoid opening these up for modification.

Reply:
@arjun4084346 , I think you can utilize the constructor to pass in the values. The understanding of `@NoArgsConstructor(force = true)` was wrong. "Force=true" doesn't force you to use the no-args constructor, the documentation says "If true, initializes all final fields to 0 / null / false" when you apply the no-args constructor.